### PR TITLE
Set new buffer limits to enable buffer size based flow control

### PR DESCRIPTION
**Pull request by sdkottegoda**
Addresses jira https://wso2.org/jira/browse/MB-1157

### CODE REVIEW

File: features/andes/org.wso2.carbon.andes.server.feature/resources/conf/broker.xml [hunk 1]
```diff
@@ -376,17 +376,16 @@ expression of the property. -->
 	 by inbound disruptor reaches below this limit	
 	-->	
     <flowControl>
-        <!-- This is the global buffer limits which enable/disable the flow control globally -->
+        <!-- These are the global buffer limits in Bytes which enable/disable the flow control globally -->
```
Comment (a5anka) [inline features/andes/org.wso2.carbon.andes.server.feature/resources/conf/broker.xml]: Should n't this be Mega bytes?

File: features/andes/org.wso2.carbon.andes.server.feature/resources/conf/broker.xml [hunk 2]
```diff
@@ -376,17 +376,16 @@ expression of the property. -->
...
         <global>
-            <lowLimit>800</lowLimit>
-            <highLimit>8000</highLimit>
+            <lowLimit>40</lowLimit>
+            <highLimit>400</highLimit>
```
Comment (a5anka) [inline features/andes/org.wso2.carbon.andes.server.feature/resources/conf/broker.xml]: Shall we add a comment on the maximum values we can use?

**Comment by ramith** [comment]
I suppose we are not going to merge this one.

**Comment by sdkottegoda** [comment]
yes

**Comment by sdkottegoda** [comment]
We have evaluated this approach vs the existing mechanism of flow controlling

With the new implementation, we face many problems since we cannot not relate between the number of messages and the total space occupied by messages at the configuration time (Messages with different sizes could be published). Following are the problems faced due to this:

    Flow controlling buffer limits are set for sizes in bytes.
        Mina buffer accepts all the messages until we enable flow controlling. (It is enabled when the total size of the messages read by the broker is higher than the specified limit.)
        For the given limit, the number of messages that could be taken in is high for small messages.
        More time is taken to read and process a large number of small messages.
        Therefore, by the time the broker enables flow controlling, the Mina buffer would have overflowed.

    The ring buffer in the disruptor is initialized with a specified buffer size with a given number of elements.
        If the publisher sends messages with smaller sizes, a large number of messages will be taken in before flow controlling is enabled, and the disruptor will overflow.
    The user cannot relate between the number of messages that are published before flow controlling is enabled (By having a count specified as in the existing approach, the number of messages that are taken in is constant whatever the message size is).

Therefore, we decided to leave the configuration as it is.

**Pull request closed without merging**